src/lib/files/format.ts

```diff
@@ -13,3 +15,36 @@ export function formatBytes(size: number) {
     return parseFloat((size / 1024 ** 3).toFixed(2)) + ' GB'
   }
 }
+
+/**
+ * Extract extension from filename.
+ * @param fileName
+ * @returns Returns `undefined` if the file has no extension
+ */
+export function extractFileExtension(fileName: string) {
+  if (!fileName.includes('.')) {
+    return
+  }
+
+  return fileName.split('.').at(-1)
+}
+
+export function formatFileExtension(extension?: string) {
+  if (extension && extension.length <= MAX_FILE_EXTENSION_DISPLAY_LENGTH) {
+    return extension.toUpperCase()
+  }
+
+  return 'File'
+}
+
+/**
+ * Extract filename omitting extension
+ * @param fileName
+ */
+export function extractFileName(fileName: string) {
+  if (!fileName.includes('.')) {
+    return fileName
+  }
+
+  return fileName.split('.').slice(0, -1).join('.')
```

Comment:
Suggestion: This is fine. May be it'd be a better approach.
```suggestion
  return fileName.replace(/\.[^.]*$/, '');
```